azure-functions/acbs-function/mappings/facility/facility-guarantee-amend.test.js

```diff
@@ -0,0 +1,56 @@
+const facilityGuaranteeAmend = require('./facility-guarantee-amend');
+
+describe('facilityGuaranteeAmend', () => {
+  const aValidAmount = 123.456;
+  const aValidAmountExpected = 123.46;
+  const aValidGuaranteeExpiryDate = '2022-01-01';
+  const aValidGuaranteeExpiryDateExpected = '2022-01-01';
+  const aValidAmountRequest = { amount: aValidAmount };
+  const aValidAmountExpectedResult = { guaranteedLimit: aValidAmountExpected };
+  const aValidGuaranteeExpiryDateRequest = { facilityGuaranteeDates: { guaranteeExpiryDate: aValidGuaranteeExpiryDate } };
+  const aValidGuaranteeExpiryDateExpectedResult = { expirationDate: aValidGuaranteeExpiryDateExpected };
+  const aValidAmountAndGuaranteeExpiryDateRequest = { ...aValidAmountRequest, ...aValidGuaranteeExpiryDateRequest };
+  const aValidAmountAndGuaranteeExpiryDateExpectedResult = { ...aValidAmountExpectedResult, ...aValidGuaranteeExpiryDateExpectedResult };
```

Comment:
Finding these hard to read - not sure if this is an established pattern, I think an object structure would be easier to read/maintain

Does it need `aValid`? Can't see any "invalid" scenarios here?

Reply:
updated to pattern.

We do have invalid tests below -- however the objects are just used for one test so no need for reusability.

Reply:
Really not a fan of these variable names they're very long e.g

```js
aValidAmountAndGuaranteeExpiryDateRequest
aValidAmountAndGuaranteeExpiryDateExpectedResult

// ... 

expect(result).toEqual(aValidAmountAndGuaranteeExpiryDateExpectedResult);
```

How about something more like this? Maybe with some kind of "generic" `expected` object

```js
const mockAmount = 123.456

const mockAmendment = {
  amount: 123.456,
  facilityGuaranteeDates: { guaranteeExpiryDate: '2022-01-01' }
};

it('should ...', () => {
  const result = facilityGuaranteeAmend(mockAmendment);

  const expected = {
    targetAmount: expected.targetAmount,
    targetAmount: 123.456,
    expirationDate: '2022-01-01',
  };

  expect(result).toEqual(expected);
});
```

Reply:
updated